pkg/dependencies/factory.go

```diff
@@ -580,6 +581,27 @@ func (f *Factory) WithProvider(clusterConfigFile string, clusterConfig *v1alpha1
 	return f
 }
 
+// WithKubeconfigReader adds the KubeconfigReader dependency depending on the provider.
+func (f *Factory) WithKubeconfigReader(clusterConfig *v1alpha1.Cluster) *Factory {
+	genericKubeconfigReader := providers.BuildKubeconfigReader(f.dependencies.UnAuthKubeClient, time.Minute, time.Second)
+	switch clusterConfig.Spec.DatacenterRef.Kind {
+	case v1alpha1.VSphereDatacenterKind:
+		f.dependencies.KubeconfigReader = genericKubeconfigReader
+	case v1alpha1.CloudStackDatacenterKind:
+		f.dependencies.KubeconfigReader = genericKubeconfigReader
+	case v1alpha1.DockerDatacenterKind:
+		f.dependencies.KubeconfigReader = docker.BuildKubeconfigReader(f.dependencies.Provider, f.dependencies.UnAuthKubeClient, time.Minute, time.Second)
+	case v1alpha1.TinkerbellDatacenterKind:
+		f.dependencies.KubeconfigReader = genericKubeconfigReader
+	case v1alpha1.SnowDatacenterKind:
+		f.dependencies.KubeconfigReader = genericKubeconfigReader
+	case v1alpha1.NutanixDatacenterKind:
+		f.dependencies.KubeconfigReader = genericKubeconfigReader
+	}
```

Comment:
Can we simplify this to
```suggestion
	switch clusterConfig.Spec.DatacenterRef.Kind {
	case v1alpha1.DockerDatacenterKind:
		f.dependencies.KubeconfigReader = docker.BuildKubeconfigReader(f.dependencies.Provider, f.dependencies.UnAuthKubeClient, time.Minute, time.Second)
	default:
		f.dependencies.KubeconfigReader = genericKubeconfigReader
	}
```